wkcuber/api/layer.py

```diff
@@ -68,25 +168,65 @@ class Layer:
     ## Functions
     """
 
-    def __init__(
-        self,
-        name: str,
-        dataset: "Dataset",
-        dtype_per_channel: np.dtype,
-        num_channels: int,
-    ) -> None:
+    def __init__(self, dataset: "Dataset", properties: LayerProperties) -> None:
         """
         Do not use this constructor manually. Instead use `wkcuber.api.layer.Dataset.add_layer()` to create a `Layer`.
         """
-        self.name = name
-        self.dataset = dataset
-        self.dtype_per_channel = dtype_per_channel
-        self.num_channels = num_channels
+        # It is possible that the properties on disk do not contain 'num_channels'.
+        # Therefore, the parameter is optional. However at this point, 'num_channels' was already inferred.
+        assert properties.num_channels is not None
+
+        self._name: str = (
+            properties.name
+        )  # The name is also stored in the properties, but the name is required to get the properties.
+        self._dataset = dataset
+        self._dtype_per_channel = _element_class_to_dtype_per_channel(
+            properties.element_class, properties.num_channels
+        )
         self._mags: Dict[Mag, MagView] = {}
 
-        full_path = join(dataset.path, name)
+        full_path = join(dataset.path, properties.name)
         makedirs(full_path, exist_ok=True)
 
+        for resolution in properties.wkw_resolutions:
+            self._setup_mag(Mag(resolution.resolution).to_layer_name())
+
+    @property
+    def _properties(self) -> LayerProperties:
+        return next(
+            layer_property
+            for layer_property in self.dataset._properties.data_layers
+            if layer_property.name == self.name
+        )
+
+    @property
+    def name(self) -> str:
+        return self._name
```

Comment:
This makes the name read-only. I think we could also add a setter ([using `@name.setter`](https://docs.python.org/3/library/functions.html#property)), since we have the reference to the dataset, and can change the dict-entry as well as the json-file.

Reply:
Do you mean we should rename the `rename()`-method to `name` (the setter)?

Reply:
Yes, exactly.